### PR TITLE
Av 23610 : Added progress bar for F5 and netscaler

### PR DESCRIPTION
**Pull request by chaitanyaavi**

### CODE REVIEW

File: python/avi/migrationtools/avi_converter.py [hunk 1]
```diff
@@ -8,6 +8,31 @@
 
 LOG = logging.getLogger(__name__)
 
+
+# Print iterations progress
+def print_progress_bar(iteration, total, msg, prefix='', suffix='', decimals=1,
+                       length=100, fill='#'):
```
Comment (grastogi23) [inline python/avi/migrationtools/avi_converter.py]: should length be 80 for 80 char window. or may be just 50 so it does not wrap around?

File: python/avi/migrationtools/config_patch.py [hunk 2]
```diff
@@ -201,12 +202,20 @@ def patch(self):
         :return:
         """
         new_cfg = deepcopy(self.avi_cfg)
+        total_size = len(self.patches)
+        progressbar_count = 0
+        print "Conversion For Patching of Object started..."
```
Comment (grastogi23) [inline python/avi/migrationtools/config_patch.py]: Patching of objects

**Review by grastogi23 — CHANGES_REQUESTED** [review]

File: python/avi/migrationtools/config_patch.py [hunk 3]
```diff
@@ -201,12 +202,20 @@ def patch(self):
         :return:
         """
         new_cfg = deepcopy(self.avi_cfg)
+        total_size = len(self.patches)
+        progressbar_count = 0
+        print "Conversion For Patching of objects started..."
+        from avi.migrationtools.avi_converter import print_progress_bar
```
Comment (grastogi23) [inline python/avi/migrationtools/config_patch.py]: why we are importing it here?

File: python/avi/migrationtools/gss_convertor/gss_parser.py [hunk 4]
```diff
@@ -30,7 +30,7 @@ def parse(val):
 
 def parser(file_name):
     """ Parsing goes here """
-    LOG.info('Parser Started')
+    LOG.info('Parserend Started')
```
Comment (grastogi23) [inline python/avi/migrationtools/gss_convertor/gss_parser.py]: spelling

**Review by grastogi23 — CHANGES_REQUESTED** [review]

File: python/avi/migrationtools/config_patch.py (hunk 3 above)
Comment (chaitanyaavi) [reply python/avi/migrationtools/config_patch.py]: There is cyclic dependency on config_patch.py and avi_converter.py hence we had to import it here.

Comment (grastogi23) [reply python/avi/migrationtools/config_patch.py]: let us then move the print_progress_bar into a common migration_utils.py. we can use this to put any common utility that is needed across different modules.

File: python/avi/migrationtools/avi_migration_utils.py [hunk 5]
```diff
@@ -0,0 +1,3324 @@
...
+    # 'ssl key and cert': {},
+    # 'ssl profile': {},
+    # 'pool group': {},
+    # 'health monitor': {},
+    # 'Httppolicy': {}
+}
+
+
+
+class Migration_Utils(object):
```
Comment (grastogi23) [inline python/avi/migrationtools/avi_migration_utils.py]: we don't use _ in the class names. so it should be MigrationUtils

File: python/avi/migrationtools/avi_migration_utils.py [hunk 6]
```diff
@@ -0,0 +1,3324 @@
...
+            format(100 * (iteration / float(total)))
+        filledLength = int(length * iteration // total)
+        bar = fill * filledLength + '-' * (length - filledLength)
+        if (iteration < total):
+            print '\r%s |%s| %s%% %s' % (prefix, bar, percent, suffix),
+        else:
+            print '\r%s |%s| %s%% %s' % (prefix, bar, percent, suffix)
+
+
+class Ns_Migration_Utils(Migration_Utils):
```
Comment (grastogi23) [inline python/avi/migrationtools/avi_migration_utils.py]: Ns_Migration_Utils -> NsMigrationUtils

**Review by grastogi23 — CHANGES_REQUESTED** [review]

**Review by grastogi23 — APPROVED** [review]

**Pull request merged**